### PR TITLE
Update changelog in prep for 7.1.0

### PR DESCRIPTION
**Pull request by rahulporuri**

### CODE REVIEW

**Comment by kitchoi** [comment]
Thanks to @rahulporuri for the initial efforts. That was very useful. I have cleaned up the summary. Realizing that it is both slow and not-particular-useful to categorize every single PR reference, I went for a highlight/excerpt approach.

**Comment by kitchoi** [comment]
Ideally, this PR should target a new `maint/7.1` branch and we can cherry pick the change log back to master after the release.

**Comment by kitchoi** [comment]
Last Travis build is failing on Wx for this sporadic error:
https://github.com/enthought/pyface/issues/556

File: CHANGES.txt [hunk 1]
```diff
@@ -2,6 +2,86 @@
...
+Detailed changes
+----------------
+
+More than 70 PRs went into this release. Thanks to:
+
+* Aaron Ayres
+* Ieva Cernyte
+* Mark Dickinson
+* Kit Yan Choi
+* Rahul Poruri
```
Comment (kitchoi) [inline CHANGES.txt:41]: @rahulporuri GitHub shows your name as "Poruri Sai Rahul", but subsequent mentions of your name in this file is "Rahul Poruri".

File: CHANGES.txt [hunk 2]
```diff
@@ -2,6 +2,86 @@
 Pyface Changelog
 ================
 
+Release 7.1.0
+=============
+
+Released: 2020-10-XX
```
Comment (kitchoi) [inline CHANGES.txt:8]: We have not been including release date. This is probably not necessary given that both PyPI and GitHub present the date attached with the release and those should be more accurate.

**Review by aaronayres35 — APPROVED** [review]
LGTM!

File: CHANGES.txt (hunk 1 above)
Comment (rahulporuri) [reply CHANGES.txt:39]: either is fine. thanks for the headsup. resolving.

**Comment by corranwebster** [comment]
> Ideally, this PR should target a new `maint/7.1` branch and we can cherry pick the change log back to master after the release.

We normally have the release PRs merge back into master after the 7.2 dev version bump and then branch from the tagged release for the maintenance branch with a 7.0.1 dev version bump.  This is easier than cherry-picking parts of the release PR.

**Comment by corranwebster** [comment]
This is the release process document for ETS: https://github.com/enthought/documentation/blob/master/source/guidelines/ets_maintenance.rst

**Pull request merged**